### PR TITLE
[Merge] Run fork choice if head block is found to be invalid

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
[Merge] Run fork choice if head block is found to be invalid

- Extended `onExecutionPayloadResult` to run fork choice if execution payload is invalid
- Modified tests accordingly

## Fixed Issue(s)
related to #5371 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java [hunk 1]
```diff
@@ -392,7 +391,9 @@ private void onExecutionPayloadResult(
     }
     transitionValidatedStatus.finishAsync(
         result -> {
-          if (result.getStatus().hasStatus(ExecutionPayloadStatus.VALID)) {
+          PayloadStatus resultStatus = result.getStatus();
```
Comment (tbenr) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java]: nit: we use `final` on local variables that are not supposed to be changed

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java [hunk 2]
```diff
@@ -402,9 +403,14 @@ private void onExecutionPayloadResult(
             }
           }
 
-          getForkChoiceStrategy()
-              .onExecutionPayloadResult(
-                  result.getInvalidTransitionBlockRoot().orElse(blockRoot), result.getStatus());
+          Bytes32 validatedBlockRoot = result.getInvalidTransitionBlockRoot().orElse(blockRoot);
```
Comment (tbenr) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java]: nit: `final` here too

**Review by tbenr — APPROVED** [review]
LGTM. Just a couple of nits

**Comment by StefanBratanov** [comment]
> LGTM. Just a couple of nits

Thanks for reminding me about the `final`. Will resolve this tomorrow.

**Review by tbenr — APPROVED** [review]
LGTM

**Pull request merged**